openpype/hosts/traypublisher/plugins/publish/collect_explicit_colorspace.py

```diff
@@ -26,18 +27,43 @@ class CollectColorspace(pyblish.api.InstancePlugin,
 
     def process(self, instance):
         values = self.get_attr_values_from_data(instance.data)
-        colorspace = values.get("colorspace", None)
-        if colorspace is None:
+        colorspace_value = values.get("colorspace", None)
+        if colorspace_value is None:
             return
 
-        self.log.debug("Explicit colorspace set to: {}".format(colorspace))
+        color_data = colorspace.convert_colorspace_enumerator_item(
+            colorspace_value, self.config_items)
+
+        colorspace_name = self._colorspace_name_by_type(color_data)
+        self.log.debug("Explicit colorspace name: {}".format(colorspace_name))
 
         context = instance.context
         for repre in instance.data.get("representations", {}):
             self.set_representation_colorspace(
                 representation=repre,
                 context=context,
-                colorspace=colorspace
+                colorspace=colorspace_name
+            )
+
+    def _colorspace_name_by_type(self, colorspace_data):
+        """
+        Returns colorspace name by type
+
+        Arguments:
+            colorspace_data (dict): colorspace data
+
+        Returns:
+            str: colorspace name
+        """
+        if colorspace_data["type"] == "colorspaces":
+            return colorspace_data["name"]
+        elif colorspace_data["type"] == "roles":
+            return colorspace_data["colorspace"]
```

Comment:
I know I've mentioned this elsewhere before but I still think it's super weird that the type is a word in plural - this makes me think that the colorspace itself refers to multiple colorspaces or roles instead of one.

Reply:
You are right, this is inherited from OCIO api. It might be confusing for some. Think of it more as a group rather than type. Lets not to change it within this PR.